examples/fit-model-ceres.cpp

```diff
@@ -18,61 +18,240 @@
+                                      const std::string &modelfile,
+                                      const std::string &outputfile) {
+            fitting::ModelContour model_contour;
+            fitting::ContourLandmarks ibug_contour;
+            if (contourfile.empty()) {
+                model_contour = fitting::ModelContour();
+                ibug_contour = fitting::ContourLandmarks();
+            } else {
+                model_contour = fitting::ModelContour::load(contourfile);
+                try {
+                    ibug_contour = fitting::ContourLandmarks::load(mappingsfile);
+                } catch (const std::runtime_error &e) {
+                    const std::string error_text =
+                            static_cast<std::string>("Error reading the contour mappings file: ") +
+                            e.what() + '\n';
+                    std::cout << error_text << std::flush;
+                    throw std::runtime_error(error_text);
+                }
+            }
+
+            Mat image = cv::imread(imagefile);
+            LandmarkCollection<Vector2f> landmarks;
+            try {
+                landmarks = core::read_pts_landmarks(landmarksfile);
+            } catch (const std::runtime_error &e) {
+                const std::string error_text = static_cast<std::string>("Error reading the landmarks: ") +
+                                               e.what() + '\n';
+                std::cout << error_text << std::flush;
+                throw std::runtime_error(error_text);
+            }
+            morphablemodel::MorphableModel morphable_model;
+            try {
+                morphable_model = morphablemodel::load_model(modelfile);
+            } catch (const std::runtime_error &e) {
+                const std::string error_text = static_cast<std::string>("Error loading the Morphable Model: ") +
+                                               e.what() + '\n';
+                std::cout << error_text << std::flush;
+                throw std::runtime_error(error_text);
+            }
+
+            // Note: Actually it's a required argument, so it'll never be empty.
+            core::LandmarkMapper landmark_mapper =
+                    mappingsfile.empty() ? core::LandmarkMapper() : core::LandmarkMapper(mappingsfile);
+
+            std::vector<morphablemodel::Blendshape> blendshapes =
+                    morphablemodel::load_blendshapes(blendshapesfile);
+
+            return FittingData {model_contour,
+                                ibug_contour,
+                                image,
+                                landmarks,
+                                morphable_model,
+                                landmark_mapper,
+                                blendshapes};
+        }
+
+        template<typename LandmarkType>
+        void draw_landmarks(Mat &image, const IndexedLandmarkCollection<LandmarkType> &landmarks,
```

Comment:
If you could please use the [.clang-format](https://github.com/patrikhuber/eos/blob/master/.clang-format) file from the repo for formatting :-) (e.g. in this particular case, the style is to put the ampersand next to the type)

Reply:
Sorry, just didn't notice `.clang-format` file.